### PR TITLE
CC3220SF: Updated SDK, XDCTools and TI Compiler

### PR DESCRIPTION
**Pull request by Palantir555**
[ Description ]

Update the tools used to compile libxively, wolfssl and the example app for the CC3220SF:

- SDK `v1.40.1.0`
- TI compmiler `v17.6.0.STS`
- XDCTOOLS `3.50.02.20`

[ JIRA ]

[ Reviewer ]
@DellaBitta

[ QA ]

#### Binaries tested:

- [x] PoC Debug w/ tls_socket
- [x] PoC Release w/ tls_socket
- [x] PoC w/ WolfSSL

#### Updating XDCTOOLS to 3.50.02.20:

- [x] libxively w/ _TLS_SOCKET builds - No changes required
- [x] libxively w/ WOLFSSL builds - No changes required

- [x] wolfssl builds - No changes required

- [x] application w/ _TLS_SOCKET builds
- [x] application w/ WOLFSSL builds

#### Updating SDK to 1.40.01.00:

- [x] libxively w/ _TLS_SOCKET builds — REQUIRED FIXES
- [x] libxively w/ WOLFSSL builds — REQUIRED FIXES

- [x] wolfssl builds

- [x] application w/ _TLS_SOCKET builds
- [x] application w/ WOLFSSL builds

#### Updating TI Compiler to 17.6.0.STS:

- [x] libxively w/ _TLS_SOCKET builds
- [x] libxively w/ WOLFSSL builds

- [x] wolfssl builds

- [x] application w/ _TLS_SOCKET builds
- [x] application w/ WOLFSSL builds

[ Release Notes ]
Updated CC3220SF Library and Build environment to use CC3220 SDK v1.4 and XDCTOOLS v3.50.02.20.

### CODE REVIEW

File: src/bsp/platform/cc3220sf/xi_bsp_io_net_cc3220sf.c [hunk 1]
```diff
@@ -4,7 +4,9 @@
  * it is licensed under the BSD 3-Clause license.
  */
 
-#include <socket.h>
```
Comment (DellaBitta) [inline src/bsp/platform/cc3220sf/xi_bsp_io_net_cc3220sf.c]: What happened here?  Did they spilt the function declarations across a series of headers?

File: make/mt-os/mt-cc3220sf.mk [hunk 2]
```diff
@@ -102,7 +102,9 @@ XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source/ti/devices/cc32xx/driverli
 XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source
 XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source/ti/drivers
 XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source/ti/drivers/net/wifi
-XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source/ti/drivers/net/wifi/sys
+XI_COMPILER_FLAGS += -I$(XI_CC3220SF_PATH_SDK)/source/ti/drivers/net/wifi/bsd
```
Comment (DellaBitta) [inline make/mt-os/mt-cc3220sf.mk]: why would hey do this?  It's like they're going back in time.

Comment (Palantir555) [inline make/mt-os/mt-cc3220sf.mk]: heh yeah, who knows....

**Comment by Palantir555** [comment]
@DellaBitta @atigyi I have now tested the application binaries. They all build and run as expected, including successful connections and sub/pubs.

PR is ready to merge when I get your 👍

File: src/bsp/platform/cc3220sf/xi_bsp_io_net_cc3220sf.c (hunk 1 above)
Comment (Palantir555) [inline src/bsp/platform/cc3220sf/xi_bsp_io_net_cc3220sf.c]: Yeah, that's it

**Pull request merged**